subprojects/core/src/main/java/org/gradle/execution/plan/ExecutionNodeAccessHierarchy.java

```diff
@@ -203,7 +218,7 @@ public File getFile() {
 
         @Override
         public boolean isDirectory() {
-            return !isFile;
+            return file.isDirectory();
```

Comment:
Is this related to this bugfix? It looks like it's more a matter of correctness, e.g symlinks, which should be tested?

Reply:
I am now using the same implementation as in `DefaultFileTreeElement`: `!relativePath.isFile()`. I don't think this matters much, though.